Content.Client/UserInterface/Controls/RadialMenu.cs

```diff
@@ -205,51 +278,308 @@ private void OnClicked(ButtonEventArgs args)
     {
         foreach (var ancestor in control.GetSelfAndLogicalAncestors())
         {
-            if (ancestor is RadialMenu)
-                return ancestor as RadialMenu;
+            if (ancestor is RadialMenu menu)
+                return menu;
         }
 
         return null;
     }
 }
 
+public interface IRadialMenuItemWithSector
+{
+    /// <summary>
+    /// Angle in radian where button sector should start.
+    /// </summary>
+    public float AngleSectorFrom { set; }
+
+    /// <summary>
+    /// Angle in radian where button sector should end.
+    /// </summary>
+    public float AngleSectorTo { set; }
+
+    /// <summary>
+    /// Outer radius for drawing segment and pointer detection.
+    /// </summary>
+    public float OuterRadius { set; }
+
+    /// <summary>
+    /// Outer radius for drawing segment and pointer detection.
+    /// </summary>
+    public float InnerRadius { set; }
+
+    /// <summary>
+    /// Offset in radian by which menu button should be rotated.
+    /// </summary>
+    public float AngleOffset { set; }
+
+    /// <summary>
+    /// Coordinates of center in parent component - button container.
+    /// </summary>
+    public Vector2 ParentCenter { set; }
+}
+
 [Virtual]
-public class RadialMenuTextureButton : TextureButton
+public class RadialMenuTextureButtonWithSector : RadialMenuTextureButton, IRadialMenuItemWithSector
 {
+    private Vector2[]? _sectorPointsForDrawing;
+
+    private float _angleSectorFrom;
+    private float _angleSectorTo;
+    private float _outerRadius;
+    private float _innerRadius;
+    private float _angleOffset;
+
+    private bool _isWholeCircle;
+    private Vector2? _parentCenter;
+
     /// <summary>
-    /// Upon clicking this button the radial menu will be moved to the named layer
+    /// Marker, that control should render border of segment. Is false by default.
     /// </summary>
-    public string TargetLayer { get; set; } = string.Empty;
+    /// <remarks>
+    /// By default color of border is same as color of background. Use <see cref="BorderColor"/>
+    /// and <see cref="HoverBorderColor"/> to change it.
+    /// </remarks>
+    public bool DrawBorder { get; set; } = false;
+
+    /// <summary>
+    /// Marker, that control should render background of all sector. Is true by default.
+    /// </summary>
+    public bool DrawBackground { get; set; } = true;
+
+    /// <summary>
+    /// Marker, that control should render separator lines.
+    /// Separator lines are used to visually separate sector of radial menu items.
+    /// Is true by default
+    /// </summary>
+    public bool DrawSeparators { get; set; } = true;
+
+    /// <summary>
+    /// Color of background in non-hovered state.
+    /// </summary>
+    public Color BackgroundColor { get; set; } = new Color(173, 216, 230, 70);
```

Comment:
```suggestion
    public Color BackgroundColor { get; set; } = new Color(70, 73, 102, 128);
```

Reply:
Leaving some suggestions to hopefully better match the exist UI color palette

Reply:
done.